saleor/product/utils.py

```diff
@@ -194,6 +192,25 @@ def get_variant_picker_data(product, discounts=None, local_currency=None):
             'schemaData': schema_data}
         data['variants'].append(variant_data)
 
+        for variant_key, variant_value in iteritems(variant.attributes):
+            filter_available_variants[int(variant_key)].append(
+                int(variant_value))
+
+    for attribute in variant_attributes:
+        availabel_variants = filter_available_variants.get(attribute.pk, None)
+
+        if not availabel_variants:
```

Comment:
Instead of checking negative case it would be more readable and shorter to check the positive case and act on it.